shared/reducers/profile.js

```diff
@@ -9,7 +9,7 @@ export default function (state = initialState, action) {
 
   switch (action.type) {
     case CommonConstants.resetStore:
-      return {...initialState}
+      return Immutable.fromJS({...initialState.toJS()})
```

Comment:
I'm sorry, That's annoying.

How about: `initialState.asMutable().asImmutable()`

Matches the property we're looking for, and will probably not subtly loose data.
```
a.asMutable().asImmutable() === a
// => false
a.asMutable().asImmutable().equals(a)
// => true
```